codenvy-api-factory/src/test/java/com/codenvy/api/factory/FactoryEditValidatorImplTest.java

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2014 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package com.codenvy.api.factory;
+
+import com.codenvy.api.account.server.dao.AccountDao;
+import com.codenvy.api.account.server.dao.Member;
+import com.codenvy.api.core.ApiException;
+import com.codenvy.api.core.ForbiddenException;
+import com.codenvy.api.core.ServerException;
+import com.codenvy.api.factory.dto.Author;
+import com.codenvy.api.factory.dto.Factory;
+
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for {@link com.codenvy.api.factory.FactoryEditValidator}
+ * @author Florent Benoit
+ */
+@Listeners(value = {MockitoTestNGListener.class})
+public class FactoryEditValidatorImplTest {
+
+    /**
+     * Account DAO
+     */
+    @Mock
+    private AccountDao accountDao;
+
+    /**
+     * Factory Mocking
+     */
+    @Mock
+    private Factory factory;
+
+    /**
+     * Instance of validator to test
+     */
+    private FactoryEditValidator factoryEditValidator;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
```

Comment:
I suppose this can be done even easier with 
@InjectMocks
private FactoryEditValidator factoryEditValidator;

instead of usage of reflection API